lib/Backend/BailOut.cpp

```diff
@@ -2430,7 +2430,16 @@ void BailOutRecord::ScheduleLoopBodyCodeGen(Js::ScriptFunction * function, Js::S
                 break;
 
             case IR::BailOutOnNotNativeArray:
-                rejitReason = RejitReason::ExpectingNativeArray;
+                // REVIEW: We have an issue with array profile info.  The info on the type of array we have won't
+                //         get fixed by rejitting.  For now, just give up after 50 rejits.
```

Comment:
I opened issue #3127
I think we're already tracking it internally, but just to be sure.
I copied the logic from what we do for function level bailouts